### PR TITLE
Release notes and related changes for release 2.3.0

### PR DESCRIPTION
**Pull request by kerry-meyer**
##### SUMMARY
This PR provides the release notes and related changes for release 2.3.0 of the enterprise_sonic Ansible collection. It is a "documentation only" PR. (No functional code changes are included.)
##### GitHub Issues
List the GitHub issues impacted by this PR. If no Github issues are affected, please indicate this with "N/A".

| GitHub Issue # |
| -------------- |
| |
N/A

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

changelog

##### OUTPUT

```paste below
N/A
```

##### ADDITIONAL INFORMATION

```paste below

```

##### Checklist:
N/A
- [ ] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [ ] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

##### How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
N/A

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,103 @@ Dellemc.Enterprise_Sonic Release Notes
...
+- sonic_ip_neighbor - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+- sonic_l2_acls - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/306).
+- sonic_l2_acls - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+- sonic_l2_interfaces - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/303).
+- sonic_l2_interfaces - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+- sonic_l3_acls - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/306).
+- sonic_l3_acls - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+- sonic_l3_interfaces - Added support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/241)
```
Comment (ArunSaravananBalachandran) [inline CHANGELOG.rst:57]:
```suggestion
- sonic_l3_interfaces - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/241).
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,103 @@ Dellemc.Enterprise_Sonic Release Notes
...
+- sonic_vrfs - Add mgmt VRF support to sonic_vrfs module (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/293).
+- sonic_vrfs - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/285).
+- sonic_vrfs - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+- tests - Add UTs for BFD, COPP, and MAC modules (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/287).
+
+Bugfixes
+--------
+
+- sonic_bgp_communities - Fix incorrect "facts" handling for parsing of a BGP community list configured with an empty "members" list (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/319).
+- sonic_bgp_neighbors - Fixed prefix-limit issue (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/289)
```
Comment (ArunSaravananBalachandran) [inline CHANGELOG.rst:92]:
```suggestion
- sonic_bgp_neighbors - Fix prefix-limit issue (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/289).
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -5,6 +5,103 @@ Dellemc.Enterprise_Sonic Release Notes
...
+- sonic_interfaces - Add warnings when speed and auto_negotiate is configured at same time (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
+- sonic_interfaces - Fix support for standard naming interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
+- sonic_interfaces - Prevent configuring speed in port group interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
+- stp - Correct the commands list for STP delete state (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/302).
```
Comment (ArunSaravananBalachandran) [inline CHANGELOG.rst:96]:
```suggestion
- sonic_stp - Correct the commands list for STP delete state (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/302).
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,103 @@ Dellemc.Enterprise_Sonic Release Notes
...
+| 4) Support for "ranges" of vlans (e.g '2-100') in tasks for the l2_interfaces and mclag resource
+| modules.
+| Please refer to the "CHANGELOG.rst" file at the top directory level of this repo for additional
+| details on the contents of this release.
+
+
+Minor Changes
+-------------
+
+- bgp - Fix the bgp CLI test base_cfg_path derivation of the bgp role_path by avoiding relative pathing from the possibly external playbook_dir (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/283).
```
Comment (ArunSaravananBalachandran) [inline CHANGELOG.rst:36]: This PR updates only the regression tests.
Hence, the module could be updated as: 
```suggestion
- tests - Fix the bgp CLI test base_cfg_path derivation of the bgp role_path by avoiding relative pathing from the possibly external playbook_dir (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/283).
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -5,6 +5,103 @@ Dellemc.Enterprise_Sonic Release Notes
...
+- regression_test - Enable contiguous execution of all regression integration tests on an S5296f (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/277).
```
Comment (ArunSaravananBalachandran) [inline CHANGELOG.rst:37]: The module can be modified to 'tests' for consistency.
```suggestion
- tests - Enable contiguous execution of all regression integration tests on an S5296f (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/277).
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -5,6 +5,103 @@ Dellemc.Enterprise_Sonic Release Notes
...
+- tests - Add UTs for BFD, COPP, and MAC modules (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/287).
+
+Bugfixes
+--------
+
+- sonic_bgp_communities - Fix incorrect "facts" handling for parsing of a BGP community list configured with an empty "members" list (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/319).
+- sonic_bgp_neighbors - Fixed prefix-limit issue (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/289)
+- sonic_interfaces - Add warnings when speed and auto_negotiate is configured at same time (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
+- sonic_interfaces - Fix support for standard naming interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
+- sonic_interfaces - Prevent configuring speed in port group interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
```
Comment (ArunSaravananBalachandran) [inline CHANGELOG.rst:95]:
```suggestion
- sonic_interfaces - Add warnings when speed and auto_negotiate is configured at same time (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314).
- sonic_interfaces - Fix support for standard naming interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314).
- sonic_interfaces - Prevent configuring speed in port group interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314).
```

**Review by ArunSaravananBalachandran — CHANGES_REQUESTED** [review]

File: changelogs/changelog.yaml [hunk 7]
```diff
@@ -516,3 +516,161 @@ releases:
...
+    changes:
+      bugfixes:
+      - sonic_bgp_communities - Fix incorrect "facts" handling for parsing of a BGP
+        community list configured with an empty "members" list (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/319).
+      - sonic_bgp_neighbors - Fixed prefix-limit issue (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/289)
+      - sonic_interfaces - Add warnings when speed and auto_negotiate is configured
+        at same time (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
+      - sonic_interfaces - Fix support for standard naming interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
+      - sonic_interfaces - Prevent configuring speed in port group interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
+      - stp - Correct the commands list for STP delete state (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/302).
```
Comment (ArunSaravananBalachandran) [inline changelogs/changelog.yaml:529]:
```suggestion
      - sonic_bgp_neighbors - Fix prefix-limit issue (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/289).
      - sonic_interfaces - Add warnings when speed and auto_negotiate is configured
        at same time (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314).
      - sonic_interfaces - Fix support for standard naming interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314).
      - sonic_interfaces - Prevent configuring speed in port group interfaces (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314).
      - sonic_stp - Correct the commands list for STP delete state (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/302).
```

File: changelogs/changelog.yaml [hunk 8]
```diff
@@ -516,3 +516,161 @@ releases:
...
+      minor_changes:
+      - bgp - Fix the bgp CLI test base_cfg_path derivation of the bgp role_path by
+        avoiding relative pathing from the possibly external playbook_dir (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/283).
+      - regression_test - Enable contiguous execution of all regression integration
+        tests on an S5296f (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/277).
```
Comment (ArunSaravananBalachandran) [inline changelogs/changelog.yaml:534]:
```suggestion
      - tests - Fix the bgp CLI test base_cfg_path derivation of the bgp role_path by
        avoiding relative pathing from the possibly external playbook_dir (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/283).
      - tests - Enable contiguous execution of all regression integration
        tests on an S5296f (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/277).
```

File: changelogs/changelog.yaml [hunk 9]
```diff
@@ -516,3 +516,161 @@ releases:
...
+      - sonic_ip_neighbor - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+      - sonic_l2_acls - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/306).
+      - sonic_l2_acls - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+      - sonic_l2_interfaces - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/303).
+      - sonic_l2_interfaces - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+      - sonic_l3_acls - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/306).
+      - sonic_l3_acls - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+      - sonic_l3_interfaces - Added support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/241)
```
Comment (ArunSaravananBalachandran) [inline changelogs/changelog.yaml:555]:
```suggestion
      - sonic_l3_interfaces - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/241).
```

File: CHANGELOG.rst [hunk 10]
```diff
@@ -5,6 +5,103 @@ Dellemc.Enterprise_Sonic Release Notes
...
+| With this release, the required support has been added for any resource modules that were not
+| provided with this support for the 2.1.0 release.
+| 2) Provide initial support for the "--check" and "--diff" mode options for playbook execution. This
+| release provides the common utility support for these options for use by all resource modules.
+| It also provides the specific resource module changes required for implementation of the
+| functionality in many of the existing resource modules. (The "--check" and "--diff" mode support
+| for the remaining resource modules is planned for inclusion in the next release.)
+| 3) New resource modules for "Public Key Infrastructure", STP, and DHCP Snooping.
+| 4) Support for "ranges" of vlans (e.g '2-100') in tasks for the l2_interfaces and mclag resource
+| modules.
```
Comment (ArunSaravananBalachandran) [inline CHANGELOG.rst:28]:
```suggestion
| 4) Support for "ranges" of vlans (e.g '2-100') in tasks for the mclag resource module.
```

File: changelogs/changelog.yaml [hunk 11]
```diff
@@ -516,3 +516,161 @@ releases:
...
+
+        | for the remaining resource modules is planned for inclusion in the next
+        release.)
+
+        | 3) New resource modules for "Public Key Infrastructure", STP, and DHCP Snooping.
+
+        | 4) Support for "ranges" of vlans (e.g ''2-100'') in tasks for the l2_interfaces
+        and mclag resource
+
+        | modules.
```
Comment (ArunSaravananBalachandran) [inline changelogs/changelog.yaml:626]:
```suggestion
        | 4) Support for "ranges" of vlans (e.g ''2-100'') in tasks for the mclag
        resource module.
```

**Review by ArunSaravananBalachandran — CHANGES_REQUESTED** [review]

**Review by ArunSaravananBalachandran — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst [hunk 12]
```diff
@@ -5,6 +5,102 @@ Dellemc.Enterprise_Sonic Release Notes
...
+- sonic_aaa - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+- sonic_acl_interfaces - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/306).
+- sonic_acl_interfaces - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+- sonic_bgp_as_paths - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/290).
+- sonic_bgp_communities - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/251).
+- sonic_bgp_ext_communities - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/252).
+- sonic_interfaces - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/301).
+- sonic_interfaces - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
```
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply CHANGELOG.rst]: Done.

File: changelogs/changelog.yaml [hunk 13]
```diff
@@ -516,3 +516,159 @@ releases:
...
+      - sonic_acl_interfaces - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/306).
+      - sonic_acl_interfaces - Enhance config diff generation function (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/318).
+      - sonic_bgp_as_paths - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/290).
+      - sonic_bgp_communities - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/251).
+      - sonic_bgp_ext_communities - Add support for replaced and overridden states
+        (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/252).
+      - sonic_interfaces - Add support for playbook check and diff modes (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/301).
+      - sonic_interfaces - Add support for replaced and overridden states (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/314)
```
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply changelogs/changelog.yaml]: Done.

File: CHANGELOG.rst (hunk 1 above)
Comment (kerry-meyer) [reply CHANGELOG.rst]: Done.

**Review by santhosh-kt — APPROVED** [review]

**Review by ArunSaravananBalachandran — APPROVED** [review]

**Pull request merged**